internal/appconfig/machines_test.go

```diff
@@ -175,6 +175,44 @@ func TestToReleaseMachineConfig(t *testing.T) {
 	assert.Equal(t, want, got)
 }
 
+func TestToTestMachineConfig(t *testing.T) {
+	cfg, err := LoadConfig("./testdata/tomachine.toml")
+	require.NoError(t, err)
+
+	want := &fly.MachineConfig{
+		Init: fly.MachineInit{
+			Cmd:        []string{"curl", "https://fly.io"},
+			SwapSizeMB: fly.Pointer(512),
+			Entrypoint: []string{"/bin/sh"},
+		},
+		Image: "curlimages/curl",
+		Env: map[string]string{
+			"PRIMARY_REGION":      "mia",
+			"FLY_TEST_COMMAND":    "1",
+			"FLY_PROCESS_GROUP":   "fly_app_test_machine_command",
+			"FLY_TEST_MACHINE_IP": "",
+			"FOO":                 "BAR",
+		},
+		Metadata: map[string]string{
+			"fly_platform_version": "v2",
+			"fly_process_group":    "fly_app_test_machine_command",
+			"fly_flyctl_version":   buildinfo.Version().String(),
+		},
+		AutoDestroy: true,
+		Restart:     &fly.MachineRestart{Policy: fly.MachineRestartPolicyNo},
+		DNS:         &fly.DNSConfig{SkipRegistration: true},
+		StopConfig: &fly.StopConfig{
+			Timeout: fly.MustParseDuration("10s"),
+			Signal:  fly.Pointer("SIGTERM"),
+		},
+	}
+
+	check := cfg.HTTPService.MachineChecks[0]
+	got, err := cfg.ToTestMachineConfig(check, nil)
```

Comment:
we have to test passing a non-nil second argument here. In fact, I don't think the second argument could be optional for the feature tested.